### PR TITLE
Added/Updated commands, created new section crypto

### PR DESCRIPTION
**Pull request by ankurparihar**
Added tree, wc, exit, gksudo, service
Updated time command
Created new section for cryptography utilities

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -369,3 +401,11 @@ ssh (SSH client) is a program for logging into a remote machine and for executin
 `-X` - Enables X11 forwarding which lets you run graphical applications remotely<br>
 `-t` - Forces a tty allocation even if a command is specified. This can be used to execute shell commands on the remote machine<br>
 `-Y` - Enables X11 connection forwarding and treats X11 clients as trusted.<br>
+
+### Cryptography
+
+`sha1sum`     &nbsp; &nbsp;     - compute and check SHA1 message digest<br/>
```
Comment (CodeMaxx) [inline README.md]: Remove the `nbsp`s

**Comment by ankurparihar** [comment]
It is done!

File: README.md [hunk 2]
```diff
@@ -369,3 +401,11 @@ ssh (SSH client) is a program for logging into a remote machine and for executin
 `-X` - Enables X11 forwarding which lets you run graphical applications remotely<br>
 `-t` - Forces a tty allocation even if a command is specified. This can be used to execute shell commands on the remote machine<br>
 `-Y` - Enables X11 connection forwarding and treats X11 clients as trusted.<br>
+
+### Cryptography
```
Comment (CodeMaxx) [inline README.md]: @ankurparihar Add some more crypto commands like `md5` and `openssl` ones. Those will be very useful, especially RSA related commands in `openssl`

Comment (ankurparihar) [reply README.md]: Added `base32`,`base64` and `md5sum` .
But I am not familiar with `openssl`. I read stuff on internet about it but so confusing.
So leaving `openssl` for now.

Comment (CodeMaxx) [reply README.md]: That is alright. It's confusing at first but very useful. I'll merge this for now. You can learn about `openssl` and send in another PR later.

File: README.md [hunk 3]
```diff
@@ -113,7 +128,9 @@ MANUAL SECTIONS
     which often include additional sections.
 ```
 
-`time <command>` - gives the time taken for the command to execute. Very useful when you want to find the execution time of your programs.
+`time <command>` - gives the time taken for the command to execute. Very useful when you want to find the execution time of your programs.<br/>
+
+`/usr/bin/time <command>` - to use format flags. Formatting output doesn't work straight away.<br/>
```
Comment (CodeMaxx) [inline README.md]: Remove this... I believe it shouldn't matter if whether you use `time` or `/usr/bin/time`

File: README.md [hunk 4]
```diff
@@ -62,12 +64,20 @@
 `cut -c 2-5 file` - cut characters 2 to 5 from each line of file<br>
 `cut -d"x" -f 1 file` - returns each part of every line before first occurrence of 'x' (-d is delimiter and -f is field)
 
+`wc [flag] filenames` - used to find out number of newline count, word count, byte and characters count in a files<br/>
```
Comment (CodeMaxx) [inline README.md]: Remove the `[flag]` part... Everyone knows how to use flags.

File: README.md [hunk 5]
```diff
@@ -13,7 +13,9 @@
 - **Important flags**<br>
 `-a` - display hidden files<br>
 `-R` - recursively displays files<br>
-`-l` - displays extra details like size, owner, group, date the file was last modified and permissions
+`-l` - displays extra details like size, owner, group, date the file was last modified and permissions<br/>
+
+`tree` - display files and directories as a tree structure.<br/>
```
Comment (CodeMaxx) [inline README.md]: This should be `tree <path>` where the default path is current directory.

File: README.md [hunk 6]
```diff
@@ -369,3 +401,22 @@ ssh (SSH client) is a program for logging into a remote machine and for executin
...
+`sha256sum` - compute and check SHA256 message digest<br/>
+`sha384sum` - compute and check SHA384 message digest<br/>
+`sha512sum` - compute and check SHA512 message digest<br/>
+
+`base32` - base32 encode/decode data and print to standard output<br/>
+`base64` - base64 encode/decode data and print to standard output<br/>
+
+**Note:** By default base32/base64 will encode data. use `-d` flag to decode data<br/>
+
+**Note:** Above mentioned commands takes file as input. To calcute hash ot text directly use `echo -n "text" | {command}`. Use of `-n` flag is mandatory. Without it, your hash will be totally wrong since it includes the newline character.<br/>
```
Comment (CodeMaxx) [inline README.md]: hash "**of**" text

Comment (CodeMaxx) [inline README.md]: The fact that the above commands take in a file as an input should be evident from the command. Use `md5sum <filename>` and so on.

**Comment by CodeMaxx** [comment]
@ankurparihar Left a few comments. Fix those and then we can merge this.

File: README.md (hunk 3 above)
Comment (ankurparihar) [reply README.md]: It does not work the normal way when we want to use formatting to get desired output.
Please have a look here : https://asciinema.org/a/0hiwdhObLo2MXkVvABxkkd8kI
Ask Ubuntu ref. : https://askubuntu.com/questions/434289/why-doesnt-the-time-command-work-with-any-option

File: README.md (hunk 5 above)
Comment (ankurparihar) [reply README.md]: corrected

File: README.md (hunk 4 above)
Comment (ankurparihar) [reply README.md]: corrected

File: README.md (hunk 6 above)
Comment (ankurparihar) [reply README.md]: corrected

File: README.md (hunk 3 above)
Comment (CodeMaxx) [reply README.md]: I see. Change that to " `/usr/bin/time <command>` - To use the "real" time command with all its flags and options. [Reference](https://askubuntu.com/questions/434289/why-doesnt-the-time-command-work-with-any-option) "

And in normal `time <command>` add that "This uses the `time` shell keyword"

Comment (ankurparihar) [reply README.md]: Ok ! done.

**Comment by CodeMaxx** [comment]
Looks good. Merged. Thanks!

**Comment by CodeMaxx** [comment]
@ankurparihar There are some sections I thought of which I've added in the Issues section. You might want to take a look.

**Pull request merged**